environment.yml

```diff
@@ -0,0 +1,12 @@
+name: pysubstrait
```

Comment:
How do we feel about leaving the conda environment as pysubstrait? Alternatively we can rename to the repo: `substrait-python`

Reply:
I have minimal opinion since I don't know what it affects. Naively, it seems most straightforward for it to match the package name

Reply:
Yeah, this you can leave alone.  A person can always override it if they want to, and it won't impact the package name on `conda-forge` (which should probably be `substrait-python` to follow convention there)